src/Operation/InsertMany.php

```diff
@@ -155,12 +155,12 @@ public function execute(Server $server)
         foreach ($this->documents as $i => $document) {
             if (isset($this->options['codec'])) {
                 $document = $this->options['codec']->encodeIfSupported($document);
-            }
 
-            // Psalm's assert-if-true annotation does not work with unions, so
-            // assert the type manually instead of using is_document
-            // See https://github.com/vimeo/psalm/issues/6831
-            assert(is_array($document) || is_object($document));
+                // Psalm's assert-if-true annotation does not work with unions, so
+                // assert the type manually instead of using is_document
+                // See https://github.com/vimeo/psalm/issues/6831
+                assert(is_array($document) || is_object($document));
```

Comment:
This has the same issue I mentioned in BulkWrite. I think it makes sense to address this in the constructor and do encoding _before_ we attempt any validation of `$document`. Interestingly, InsertOne _does not_ have the same problem and applies the codec in its constructor.

I'll confess that this is a departure from our existing convention of dealing with named arguments prior to anything in `$options`, but I think there's a good reason for doing so in this case.